src/hal/src/lib.rs

```diff
@@ -306,6 +306,7 @@ pub trait Instance {
 
 /// Different types of a specific API.
 #[allow(missing_docs)]
+#[cfg(feature = "single-threaded")]
```

Comment:
It would be great to avoid duplication of this mega-trait.
I wonder if we could have
```rust
trait Threaded: Send + Sync {
}
trait PhysicalDevice<B>: Threaded {
...
}
```
and such, so that the only thing we need to replace for `single-threaded` is this `Threaded` trait

Reply:
ping again